File: graphs/123.cpp
#include <iostream>
using namespace std;
using ll = long long;

const ll MOD = 998244353;

// Function to compute x^y % MOD using binary exponentiation
ll mod_pow(ll x, ll y, ll mod) {
    ll res = 1;
    while (y > 0) {
        if (y % 2 == 1)
            res = res * x % mod;
        x = x * x % mod;
        y /= 2;
    }
    return res;
}

// Function to compute modular inverse using Fermat's Little Theorem
ll mod_inverse(ll x, ll mod) {
    return mod_pow(x, mod - 2, mod);
}

// Function to compute the expected number of days
ll expected_days(ll W, ll G, ll L) {
    if (L == 0) {
        // If L = 0, we can only lose weight every day
        return (W - G) % MOD;
    }

    // Probabilistic geometric series for random walk
    // Expected number of days = 2 * (W - G + 1)
    ll numerator = 2 * (W - G + 1) % MOD;
    return numerator % MOD;
}

int main() {
    int T;
    cin >> T;
    for (int t = 1; t <= T; ++t) {
        ll W, G, L;
        cin >> W >> G >> L;
        ll result = expected_days(W, G, L);
        cout << "Case #" << t << ": " << result << endl;
    }
    return 0;
}

File: 123.py
MOD = 998244353

# Function to compute x^y % MOD
def mod_pow(x, y, mod):
    res = 1
    while y > 0:
        if y % 2 == 1:
            res = res * x % mod
        x = x * x % mod
        y //= 2
    return res

# Function to compute the modular inverse of a number using Fermat's Little Theorem
def mod_inverse(x, mod):
    return mod_pow(x, mod - 2, mod)

# Function to calculate the expected number of days
def expected_days(W, G, L):
    if L == 0:
        # If L = 0, we can never gain weight, so it takes exactly W - G days
        return (W - G) % MOD
    else:
        # Probabilistic geometric series for random walk with a limit
        # Expected number of days = 2 * (W - G + 1)
        numerator = 2 * (W - G + 1) % MOD
        denominator = 1  # For p/q = numerator / denominator, denominator is 1
        return numerator % MOD

# Main function to solve the problem
def solve():
    T = int(input())
    for t in range(1, T + 1):
        W, G, L = map(int, input().split())
        result = expected_days(W, G, L)
        print(f"Case #{t}: {result}")

# Input and execution
solve()
